### ISSUE TITLE
how to reconstruct data after replacing a whole disk?

### ISSUE THREAD

**Issue by nullne** [minio/minio#7785]
**Is your feature request related to a problem? Please describe.**
the normal heal operation is quite slow and unnecessary when one disk is replaced. it's better only to fill the lost data on this disk. 

**Describe the solution you'd like**
scan to find out all missing objects and only write to the missing disk.

**Describe alternatives you've considered**
now we heal the whole cluster very slowly.
**Additional context**

**Comment by nitisht** [comment]
@nullne we now support self heal (ref #7604), this will scan the drives on a daily basis and heal only the objects that need healing on certain drives. 

If you want to trigger heal your self, you can use https://docs.min.io/docs/minio-admin-complete-guide.html#heal

**Comment by nullne** [comment]
@nitisht i'm afraid i didn't make it clear. what i talked about is how to quickly recover from disk failure. the normal heal operation will scan the whole cluster and heal the object one by one. more  specifically, let's say we have 10 drives per set, the normal heal operation will read from 10 drives and compare the corresponding data. in my scenario(one disk is replaced with a brand new one), i'd like to cover the data on this disk as soon as possible rather than the whole cluster(maybe there's some objects need to be healed too).

**Comment by kannappanr** [comment]
Won't normal healing take care of this? You still need to know what all objects need to go into this drive. If there are more than one set, we can maybe look at healing one particular set that this disk belongs to.

**Comment by nullne** [comment]
@kannappanr yeah, healing one particular set is a big improve. as to normal healing, it does work on this case. but normal healing does more things (healing objects on other disks) which is not the first priority, and certainly slow the healing process

**Comment by harshavardhana** [comment]
Object names themselves indicate which set they belong to. The enhancement that may be possible in healing is to prioritize healing of objects which belong to the set where the disk was replaced.

**Comment by kannappanr** [comment]
Yes, but I don't think it should be healing just this disk, if there are missing parts in other disks, then it should heal them also.

**Comment by harshavardhana** [comment]
It should be prioritized, an unformatted disk (replaced) disk takes priority.

**Comment by kannappanr** [comment]
Yes, but should you skip other disks in the same set, even though they might have parts missing?

**Comment by harshavardhana** [comment]
It won't skip, they are just lower priority. Healing per object per disk is serial and heals one disk at a time. All it needs to do is heal the unformatted first. Since we heal objects in parallel, at top level invariably this should fill the unformatted disk quickly. 

Current implementation lacks any sort of ordering.

**Comment by harshavardhana** [comment]
Fixed in #8114
